ledger/src/leader_schedule.rs

```diff
@@ -48,6 +64,28 @@ impl LeaderSchedule {
     pub fn num_slots(&self) -> usize {
         self.slot_leaders.len()
     }
+
+    /// Returns indices starting from 'offset' where pubkey is the leader.
+    pub(crate) fn get_schedule(
+        &self,
+        pubkey: &Pubkey,
+        offset: usize, // Starting index.
+    ) -> impl Iterator<Item = usize> {
+        let index = self.index.get(pubkey).cloned().unwrap_or_default();
+        let num_slots = self.slot_leaders.len();
+        let size = index.len();
+        #[allow(clippy::reversed_empty_ranges)]
+        let range = if index.is_empty() {
+            1..=0 // Intentionally empty range of type RangeInclusive.
+        } else {
+            let offset = index
+                .binary_search(&(offset % num_slots))
+                .unwrap_or_else(identity)
+                + offset / num_slots * size;
+            offset..=usize::MAX
+        };
+        range.map(move |k| index[k % size] + k / size * num_slots)
```

Comment:
nits: Maybe add a comment for the `k / size * num_slots` because I think it's primary use is also future proofing. Maybe something like:

If we want to know the `pubkey`'s `k`th leader slot, and `k` > `num_pubkey_owned_leader_indexes`, that means we've  already gone through `k/num_pubkey_owned_leader_indexes` iterations of the entire leader schedule. 
This means `pubkey_owned_leader_schedule_indexes[0]` is an offset from the `k/num_pubkey_owned_leader_indexes` *  total_leader_schedule_len` slots that have already passed since the last time a complete round of the leader schedule was finished.

Reply:
Added more comments, but again this is just replicating existing logic:
https://github.com/solana-labs/solana/blob/ab0f4c69a/ledger/src/leader_schedule.rs#L53-L59